db/external_sst_file_test.cc

```diff
@@ -1345,15 +1345,18 @@ TEST_F(ExternalSSTFileTest, PickedLevelBug) {
   ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->EnableProcessing();
 
   // While writing the MANIFEST start a thread that will ask for compaction
+  Status bg_compact_status;
```

Comment:
I am not certain, but... to avoid stale reads due to CPU caching etc, doesn't `bg_compact_status` need synchronisation as it is written and read from different threads? I would have thought that `std::atomic` is needed here.

Same goes for `bg_addfile_status` below...

Reply:
One thread per field -> no write/write conflict
Thread join before read -> no read/write conflict

Reply:
> Thread join before read -> no read/write conflict

To use the term from the standard, the thread join creates a *synchronizes-with* relationship (see https://en.cppreference.com/w/cpp/thread/thread/join).